src/promptflow/promptflow/_sdk/entities/_connection.py

```diff
@@ -748,6 +855,42 @@ def _from_mt_rest_object(cls, mt_rest_obj):
             last_modified_date=mt_rest_obj.last_modified_date,
         )
 
+    def is_custom_strong_type(self):
+        return self.custom_type is not None
+
+    def convert_to_custom_strong_type_connection(self):
+        module_name = self.configs.get(CustomStrongTypeConnectionConfigs.FULL_MODULE)
+        custom_type_class_name = self.configs.get(CustomStrongTypeConnectionConfigs.FULL_TYPE)
+        import importlib
+
+        module = importlib.import_module(module_name)
+        custom_defined_connection_class = getattr(module, custom_type_class_name)
+
+        instance_dict = {}
+        for key, value in self.configs.items():
+            if key not in [CustomStrongTypeConnectionConfigs.FULL_MODULE, CustomStrongTypeConnectionConfigs.FULL_TYPE]:
+                instance_dict[key] = value
+        for key, value in self.secrets.items():
+            instance_dict[key] = value
+        connection_instance = custom_defined_connection_class(**instance_dict)
+
+        return connection_instance
+
+    @classmethod
+    def convert_strong_type_to_custom(cls, custom_str_type_connection: _Connection):
+        attributes = copy.copy(vars(custom_str_type_connection))
+        attributes["module"] = PROMPTFLOW_CONNECTIONS
+        # update configs
+        configs = {}
+        configs.update({CustomStrongTypeConnectionConfigs.FULL_TYPE: custom_str_type_connection.__class__.__name__})
+        configs.update({CustomStrongTypeConnectionConfigs.FULL_MODULE: custom_str_type_connection.__module__})
+        configs.update(**custom_str_type_connection.configs)
+
+        attributes["configs"] = configs
```

Comment:
is it possible to separate configs & secrets like the original custom connection? As secrets will be scrubbed in logs/std outputs, but configs will not be scrubbed which may cause secrets exposed.

Reply:
Offline synced, custom_str_type_connection has configs and secrets itself